core/modules/wsched/GroupScheduler.cc

```diff
@@ -44,127 +44,109 @@
 namespace lsst {
 namespace qserv {
 namespace wsched {
-////////////////////////////////////////////////////////////////////////
-// class GroupScheduler
-////////////////////////////////////////////////////////////////////////
-GroupScheduler::GroupScheduler()
-    : _maxRunning(4), // FIXME: set to some multiple of system proc count.
-      _logger(LOG_GET(getName())) {
-}
 
-struct matchHash {
-    explicit matchHash(std::string const& hash_) : hash(hash_) {}
-    bool operator()(wbase::Task::Ptr const& t) {
-        return t && (t->hash == hash);
-    }
-    std::string const hash;
-};
 
-bool
-GroupScheduler::removeByHash(std::string const& hash) {
-    int removed = _queue.removeIf(matchHash(hash));
-    return removed > 0;
+GroupQueue::GroupQueue(int maxAccepted, wbase::Task::Ptr const& task) : _maxAccepted{maxAccepted} {
+    assert(task != nullptr);
+    _hasChunkId = task->msg->has_chunkid();
+    if (_hasChunkId) {
+        _chunkId = task->msg->chunkid();
+    }
+    assert(queTask(task));
 }
 
-void
-GroupScheduler::queueTaskAct(wbase::Task::Ptr incoming) {
-    std::lock_guard<std::mutex> guard(_mutex);
-    _enqueueTask(incoming);
+/// Return true if this GroupQueue accepts this Task.
```

Comment:
Given the line below, I think Task should be consistently capitalized one way or the other.